src/bootstrap/compile.rs

```diff
@@ -346,6 +376,34 @@ impl Step for Test {
 
         builder.ensure(Std { compiler, target });
 
+        if compiler.stage == 0 {
+            // Copy the libtest dynamic libraries from the bootstrap compiler
+            let out_dir = build.cargo_out(compiler, Mode::Libtest, target);
+            let base_libdir = builder.config
+                                     .initial_rustc.parent().unwrap()
+                                     .parent().unwrap()
+                                     .join("lib").join("rustlib").join(compiler.host).join("lib");
+            t!(fs::create_dir_all(&out_dir));
+            let mut files: Vec<PathBuf> = Vec::new();
+            for f in t!(fs::read_dir(&base_libdir)).map(|f| t!(f)) {
+                let filename = f.file_name().into_string().unwrap();
+                if (filename.starts_with("test-") || filename.starts_with("libtest-")
+                   || filename.starts_with("term-") || filename.starts_with("libterm-")) &&
```

Comment:
This looks almost the same as it is for `libstd`, can these be refactored?

Reply:
I refactored it out to a function.